### ISSUE TITLE
Improvement: Command executed on iPad (and iPhone)

### ISSUE THREAD

**Issue by heitbaum** [xbmc/Official-Kodi-Remote-iOS#579]
@wutschel - I was thinking that the `Command executed` dialog box is a bit “DEBUG”. In Kodi itself it pops up when cleaning - then goes away. And Scanning just updates the top right as it is scanning. 

I don’t know the right way to do the user feedback. But maybe the blue clockwise moving snake around the icon (like in Termius.) 

Potentially colour in the icon “if completed” is a state (not sure how that might relate to these Command - but an idea) … grey —> blue…

### the Commands as in the iPad.
![ED08F79E-36BE-49BB-9F71-6E24D167399B](https://user-images.githubusercontent.com/6086324/150953277-b5990432-c5f6-4c9c-a5eb-fd1a2b605eaf.jpeg)
### The “DEBUG” button.
![E6916D8B-1ACB-46A9-880A-D8C1A967677B](https://user-images.githubusercontent.com/6086324/150953290-605ec4cb-8b81-4437-ac0c-73eea3688f91.jpeg)
### The Termius “action activity”
![8091EA81-FB3C-4BAC-9B7A-2EF003A4E5E3](https://user-images.githubusercontent.com/6086324/150954928-e812a48c-6967-4c27-90f9-8ab6db916862.jpeg)

**Comment by wutschel** [comment]
True, the same response is given for any other `Input.Action` executed from the Kodi action list. For the lower 4 commands Kodi is even sharing start/finish notifications which can be used for any kind of visualization. I guess this was not implemented in the App initially as this requires Kodi API 6 or later, which came with Kodi 12. The App supports down to Kodi 11. Personally I would like to drop support for Kodi 11, which would also make life easier for this change.

On visualization: On iPad there are some options to place (animated?) icons in the toolbar. For iPhone this is difficult. In the interest of the complexity I would also simply go for a popup which shows something like "Cleaning library ..." and when finished closes this automatically. Alternatively "Cleaning finished." can be shown with a timeout. The user should be offered the OK button to close the popup manually, if this is desired.

**Comment by heitbaum** [comment]
Definitely understand the challenges there on support of old / new. I like your simple popup idea with timer+ok for K11 and if feasible start-stop+ok for K12+ 

you are right about the screen space though - one of the reasons I put iPhone in () in the title. I use `actions` much less on the iPhone because they are inconvenient to get to.

I use both devices, whilst the others in the house use their iPhones with Kodi.

**Comment by wutschel** [comment]
Would you recommend to also use popup with timer + OK for the other actions (the ones hidden well behind the remote screen)?

**Comment by wutschel** [comment]
I am just reading this again while I am working on the "Power" menu. I am now unifying the way the feedback ("Executed" or "Could not do that") is shown in the UI. On iPad and iPhone this will now use the toaster animation which slides in in green (execution success) or red (execution failure) from the top for a few seconds. In addition, on iPad selecting an action from the power menu now does bring up the same popup as on iPhone which asked to reconfirm, if you really want to do this. I feel this is needed as all actions related to either the power state of Kodi or working on the database.

You think this will also resolve the issue you raised?

**Comment by heitbaum** [comment]
![IMG_2379](https://github.com/xbmc/Official-Kodi-Remote-iOS/assets/6086324/ae28e3ae-c4ac-415e-a8f6-7b587662b67a)
![IMG_2381](https://github.com/xbmc/Official-Kodi-Remote-iOS/assets/6086324/6eb87489-3b78-4da3-9342-7570426b8763)

I was just thinking about this the other night when using the iPad. I like how the iPhone works. So the toast should be fine.

**Comment by wutschel** [comment]
This is how it would like: [video iPad simulator]( https://www.dropbox.com/scl/fi/20v4y8wc4xdbel1czn8ew/Simulator-Screen-Recording-iPad-8th-generation-2024-03-20-at-12.31.13.mp4?rlkey=1ov0rh43bddhtorha40lvjiam&dl=0)

**Comment by heitbaum** [comment]
> This is how it would like: [video iPad simulator](https://www.dropbox.com/scl/fi/20v4y8wc4xdbel1czn8ew/Simulator-Screen-Recording-iPad-8th-generation-2024-03-20-at-12.31.13.mp4?rlkey=1ov0rh43bddhtorha40lvjiam&dl=0)

Looks good. Same look and feel as the phone and not “debug” like. 

Only pedantic note - is should the buttons be next to each other same as iPhone? (I have no real preference - but same ux across devices is usual the right way)

![IMG_0256](https://github.com/xbmc/Official-Kodi-Remote-iOS/assets/6086324/1955ca9a-9051-47ec-bde2-482798ac14cc)

**Comment by wutschel** [comment]
This is automatically done. It looks same on both iPad and iPhone, if the text for the buttons is same. The German "Audio aktualisieren" is too long to place it next to "Abbrechen". The English "Update audio" is placed right of "Cancel" using the same code.

**Comment by heitbaum** [comment]
> This is automatically done. It looks same on both iPad and iPhone, if the text for the buttons is same. The German "Audio aktualisieren" is too long to place it next to "Abbrechen". The English "Update audio" is placed right of "Cancel" using the same code.
> 
> Screenshot: https://ibb.co/wywNn7y

that makes sense. Looks good

**Comment by heitbaum** [comment]
Tested 1.15 4339 - I like it - looks really good.

**Comment by wutschel** [comment]
> Tested 1.15 4339 - I like it - looks really good.

Thanks, glad you like it. :)
